x-pack/legacy/plugins/ml/public/application/jobs/jobs_list/components/jobs_list_view/jobs_list_view.js

```diff
@@ -90,59 +79,18 @@ export class JobsListView extends Component {
     }
   }
 
-  componentWillUnmount() {
-    if (this.props.isManagementTable === undefined) {
-      if (this.refreshIntervalSubscription) this.refreshIntervalSubscription.unsubscribe();
-      deletingJobsRefreshTimeout = null;
-      this.clearRefreshInterval();
-    }
-  }
-
-  initAutoRefresh() {
-    const { value } = timefilter.getRefreshInterval();
-    if (value === 0) {
-      // the auto refresher starts in an off state
-      // so switch it on and set the interval to 30s
-      timefilter.setRefreshInterval({
-        pause: false,
-        value: DEFAULT_REFRESH_INTERVAL_MS,
-      });
-    }
-
-    this.setAutoRefresh();
-  }
-
-  initAutoRefreshUpdate() {
-    // update the interval if it changes
-    this.refreshIntervalSubscription = timefilter.getRefreshIntervalUpdate$().subscribe({
-      next: () => this.setAutoRefresh(),
-    });
-  }
-
-  setAutoRefresh() {
-    const { value, pause } = timefilter.getRefreshInterval();
-    if (pause) {
-      this.clearRefreshInterval();
-    } else {
-      this.setRefreshInterval(value);
+  componentDidUpdate(prevProps) {
+    if (prevProps.lastRefresh !== this.props.lastRefresh && this.props.blockRefresh === false) {
```

Comment:
as `blockRefresh ` is optional, should this check be for `!== true`?
do we want to treat a non-existing `blockRefresh` as being a false condition? if that makes sense.

Reply:
Fixed in c8ec7a5